docs/windows.md

```diff
@@ -0,0 +1,158 @@
+    - [Platform requirements](#platform-requirements)
+    - [Setup Instructions](#setup-instructions)
+    - [Example Build](#example-build)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Experimental Windows containers support
+
+As from `v0.13`, Buildkit now has exprimental support for Windows containers (WCOW). Both `buildctl.exe` and `buildkitd.exe` binaries are being released for testing purposes.
+
+We will apprecate any feedback by [opening an issue here](https://github.com/moby/buildkit/issues/new), as we stabilize the product, especially `buildkitd.exe`.
+
+## Quick start guide
+
+### Platform requirements
+
+- Architecture: `amd64`, `arm64` (binaries available but not officially tested yet).
+- Supported OS: Windows Server 2019, Windows Server 2022, Windows 11.
+- Base images: `ServerCore:ltsc2019`, `ServerCore:ltsc2022`, `NanoServer:ltsc2022`. See the [compatibility map here](https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility?tabs=windows-server-2019%2Cwindows-11#windows-server-host-os-compatibility).
+
+### Setup Instructions
+
+**Dependency:** `containerd` v1.7.7+
+
+> **NOTE:** all these requires running as admin (elevated) on a PowerShell terminal.
+
+1. Setup `containerd` by following [the setup instructions here](https://github.com/containerd/containerd/blob/main/docs/getting-started.md#installing-containerd-on-windows). (_Currently, we only support the `containerd` worker_.)
+1. Start the `containerd` service, if not yet started.
+1. Download and extract:
+    ```powershell
+    $version = "v0.13.0-rc2" # specify the release version, v0.13+
+    $arch = "amd64" # arm64 binary available too
+    curl.exe -LO https://github.com/moby/buildkit/releases/download/$version/buildkit-$version.windows-$arch.tar.gz
+    # there could be another `.\bin` directory from containerd instructions
+    # you can move those
+    mv bin bin2
+    tar.exe xvf .\buildkit-$version.windows-$arch.tar.gz
+    ## x bin/
+    ## x bin/buildctl.exe
+    ## x bin/buildkitd.exe
+    ```
+1. Setup `buildkit` binaries:
+    ```powershell
+    # after the binaries are extracted in the bin directory
+    # move them to an appropriate path in your $Env:PATH directories or:
+    Copy-Item -Path ".\bin" -Destination "$Env:ProgramFiles\buildkit" -Recurse -Force
+    # add `buildkitd.exe` and `buildctl.exe` binaries in the $Env:PATH
+    $Path = [Environment]::GetEnvironmentVariable("PATH", "Machine") + `
+        [IO.Path]::PathSeparator + "$Env:ProgramFiles\buildkit"
+    [Environment]::SetEnvironmentVariable( "Path", $Path, "Machine")
+    $Env:Path = [System.Environment]::GetEnvironmentVariable("Path","Machine") + ";" + `
+        [System.Environment]::GetEnvironmentVariable("Path","User")
+    ```
+
+    **Alternatively**, for combined steps 3 and 4, you can use the following script. Remember to specify the correct version:
```

Comment:
Sorry, I didn't mean duplicating the script. In "Download and extract:" step, the script could be optional and in here it can just be "Setup `buildkit` binaries"